AzureMonitorAgent/ama_tst/modules/general_health/general_health.py

```diff
@@ -0,0 +1,72 @@
+import os
+
+from error_codes  import *
+from errors       import error_info, is_error, print_errors
+from .check_status import check_restart_status
+
+ERR_FILE_PATH = "/var/opt/microsoft/azuremonitoragent/log/mdsd.err"
+
+def check_err_file():
+    """
+    output mdsd.err contents if the file is not empty
+    """
+    tail_size = -50
+    pattern = ' [DAEMON] '
+    err_logs = []
+    with open(ERR_FILE_PATH) as f:
+        lines = f.readlines()
+        lines = lines[tail_size:]
+        for line in lines:
+            line = line.rstrip('\n')
+            # skip empty lines, daemon start/exit logs
+            if line == '':
+                continue
+            elif pattern in line:
+                continue
+            else:
+                err_logs.append(line)
+                
+    if len(err_logs) > 0:
+        err_logs_str = '\n' + ('\n'.join(err_logs))
+        error_info.append((ERR_FILE_PATH, err_logs_str))
+        return WARN_MDSD_ERR_FILE
+    return NO_ERROR
+
+
+def ask_restart_ama():
+    ama_dir = filter((lambda x : x.startswith("Microsoft.Azure.Monitor.AzureMonitorLinuxAgent-")), os.listdir("/var/lib/waagent"))
+    ama_dir = list(ama_dir)
+    if len(ama_dir) > 1:
+        return ERR_MULTIPLE_AMA
+    print("If you need to restart Azure Monitor Agent on this machine, please execute the following commands:\n")
```

Comment:
does the customer have enough context on whether or not they need to restart? it might be difficult to provide such context (e.g. for which errors do we think restarting will actually solve the issue), but without some general guidance customer might be confused as to whether restart is necessary for them.

Reply:
Add restart instructions when user exits TST, since we don't have restart instructions in public doc (we only have restart instructions in CSS doc).